src/bbs2gh/Commands/MigrateRepoCommand.cs

```diff
@@ -44,98 +38,92 @@ public class MigrateRepoCommand : Command
+        public async Task Invoke(MigrateRepoCommandArgs args)
         {
-            _log.Verbose = verbose;
+            if (args is null)
+            {
+                throw new ArgumentNullException(nameof(args));
+            }
+
+            _log.Verbose = args.Verbose;
 
             _log.LogInformation("Migrating Repo...");
-            _log.LogInformation($"ADO ORG: {adoOrg}");
-            _log.LogInformation($"ADO TEAM PROJECT: {adoTeamProject}");
-            _log.LogInformation($"ADO REPO: {adoRepo}");
-            _log.LogInformation($"GITHUB ORG: {githubOrg}");
-            _log.LogInformation($"GITHUB REPO: {githubRepo}");
-            if (ssh)
-            {
-                _log.LogWarning("SSH mode is no longer supported. --ssh flag will be ignored.");
-            }
-            if (wait)
+            _log.LogInformation($"GITHUB ORG: {args.GithubOrg}");
+            _log.LogInformation($"GITHUB REPO: {args.GithubRepo}");
+
+            if (args.GithubPat is not null)
             {
-                _log.LogInformation("WAIT: true");
+                _log.LogInformation("GITHUB PAT: ***");
             }
-            if (adoPat is not null)
+            if (args.GithubApiUrl is not null)
             {
-                _log.LogInformation("ADO PAT: ***");
+                _log.LogInformation($"GITHUB API URL: {args.GithubApiUrl}");
             }
-            if (githubPat is not null)
+            if (args.Wait)
             {
-                _log.LogInformation("GITHUB PAT: ***");
+                _log.LogInformation("WAIT: true");
             }
 
-            githubPat ??= _environmentVariableProvider.GithubPersonalAccessToken();
-            var githubApi = _githubApiFactory.Create(targetPersonalAccessToken: githubPat);
-
-            var adoRepoUrl = GetAdoRepoUrl(adoOrg, adoTeamProject, adoRepo);
-
-            adoPat ??= _environmentVariableProvider.AdoPersonalAccessToken();
-            var githubOrgId = await githubApi.GetOrganizationId(githubOrg);
-            var migrationSourceId = await githubApi.CreateAdoMigrationSource(githubOrgId, null);
+            args.GithubPat ??= _environmentVariableProvider.GithubPersonalAccessToken();
+            var githubApi = _githubApiFactory.Create(apiUrl: args.GithubApiUrl, targetPersonalAccessToken: args.GithubPat);
+            var githubOrgId = await githubApi.GetOrganizationId(args.GithubOrg);
+            var migrationSourceId = await githubApi.CreateBbsMigrationSource(githubOrgId);
 
             string migrationId;
 
             try
             {
-                migrationId = await githubApi.StartMigration(migrationSourceId, adoRepoUrl, githubOrgId, githubRepo, adoPat, githubPat);
+                migrationId = await githubApi.StartMigration(
```

Comment:
could we move the knowledge about not-used stuff down into githubApi class. For example, maybe have a 2nd overload of startMigration that doesn't have args for the 3 not-used things, then in that function put the dummy values in the api payload.

Reply:
Done!

- Added `StartBbsMigration` to `GithubApi` with 88113db3bd988c43deced584512c2c0548ff65b1
- Made `MigrateRepoCommand` call `StartBbsMigration` with cce2213d0a3df9a176c660b5db27b8327d5e057c